rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -304,10 +297,236 @@ namespace rocal{
+                .def(
+                "batch_height",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().max_shape().at(1);
+                },
+                R"code(
+                Returns a tensor buffer's height.
+                )code"
+            )
+            .def(
+                "batch_width",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().max_shape().at(0);
+                },
+                R"code(
+                Returns a tensor buffer's width.
+                )code"
+            )
+            .def(
+                "batch_size",
+                [](rocalTensor &output_tensor)
+                {
+                    return output_tensor.info().dims().at(0);
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def(
+                "color_format",
+                [](rocalTensor &output_tensor)
+                {
+                    switch(output_tensor.info().color_format())
+                    {
+                        case RocalColorFormat::RGB24:
+                            return 3;
+                        case RocalColorFormat::BGR24:
+                            return 3;
+                        case RocalColorFormat::RGB_PLANAR:
+                            return 3;
+                        case RocalColorFormat::U8:
+                            return 1;
+                    }
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def("layout", [](rocalTensor &output_tensor)
+            {
+                switch(output_tensor.info().layout()) {
+                    case RocalTensorlayout::NHWC:
+                        return "NHWC";
+                    case RocalTensorlayout::NCHW:
+                        return "NCHW";
+                }
+            },
+                R"code(
+                Returns layout of tensor.
+                )code"
+            )
+            .def("dtype", [](rocalTensor &output_tensor)
+            {
+                switch(output_tensor.info().data_type()) {
+                    case RocalTensorDataType::FP32:
+                        return "float32";
+                    case RocalTensorDataType::UINT8:
+                        return "uint8";
+                    case RocalTensorDataType::FP16:
+                        return "float16";
+                }
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def(
+            "copy_data", [](rocalTensor &output_tensor, py::object p)
+            {
+                auto ptr = ctypes_void_ptr(p);
+                output_tensor.copy_data(ptr);
+            }
+            )
+            .def(
+                "at",
```

Comment:
This part should work not only for the images but also for the NFHWC types and for Audio as well, we need to make it generic

Reply:
Would be added later for Video & Audio Datas